src/Parser/TIFFParser.js

```diff
@@ -0,0 +1,74 @@
+import * as UTIF from 'utif';
+import * as THREE from 'three';
+
+/**
+ * The TIFFParser module provides a [parse]{@link module:TIFFParser.parse}
+ * method that takes a TIFF in and gives a `THREE.DataTexture` that can be
+ * displayed in the view.
+ *
+ * It uses the [utif](https://www.npmjs.com/package/utif) library to parse the
+ * TIFF.
+ *
+ * @example
+ * Fetcher.arrayBuffer('image.tiff')
+ *     .then(TIFFParser.parse)
+ *     .then(function _(texture) {
+ *         var source = new FileSource({ parsedData: texture });
+ *         var layer = new ColorLayer('tiff', { source });
+ *         view.addLayer(layer);
+ *     });
+ *
+ * @module TIFFParser
+ */
+export default {
+    /**
+     * Parse a TIFF file and return a `THREE.DataTexture`.
+     *
+     * @param {ArrayBuffer} data - The TIFF file content to parse.
+     *
+     * @return {Promise} A promise resolving with a `THREE.DataTexture`.
+     */
+    parse(data) {
+        const IFD = UTIF.decode(data)[0];
+        UTIF.decodeImage(data, IFD);
+        IFD.data = UTIF.toRGBA8(IFD);
+
+        // A 4096 limitation sounds the most probable here
+        // https://webglstats.com/webgl/parameter/MAX_TEXTURE_SIZE
+        if (IFD.width > 4096 || IFD.height > 4069) {
+            throw new Error('The loaded TIFF is too big: please reduce it to a maximum size of 4096 by 4096');
+        }
+
+        // Round to next power of two
+        const width = THREE.Math.ceilPowerOfTwo(IFD.width);
+        const height = THREE.Math.ceilPowerOfTwo(IFD.height);
+
+        let resizedData;
+        if (width == IFD.width && height == IFD.height) {
+            resizedData = IFD.data;
+        } else {
+            resizedData = new IFD.data.constructor(width * height * 4);
```

Comment:
THREE use canvas to resize image see : https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLTextures.js#L63

Reply:
Yes but then we got a warning: I want to avoid that.

Reply:
yes, but may be is better, to resizing, to use `canva.context.drawImage( image, 0, 0, width, height )`.

Reply:
I tried it: it is too complicated IMO, because `drawImage` can't take a `TypedArray` as an argument, so I first need to convert the `TypedArray` to an image, then draw it again to resized it. I prefer to read it manually and put it into a `THREE.DataTexture`.